File: frontend/src/app/controller/service/admin/gestionchequeffet/ProprietaireAdmin.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {BehaviorSubject, Observable} from 'rxjs';

import { RoleService } from 'src/app/zynerator/security/Role.service';
import {environment} from 'src/environments/environment';

import {ProprietaireDto} from 'src/app/controller/model/gestionchequeffet/Proprietaire.model';
import {ProprietaireCriteria} from 'src/app/controller/criteria/gestionchequeffet/ProprietaireCriteria.model';
import {AbstractService} from 'src/app/zynerator/service/AbstractService';


@Injectable({
  providedIn: 'root'
})
export class ProprietaireAdminService extends AbstractService<ProprietaireDto, ProprietaireCriteria> {
     constructor(private http: HttpClient, private roleService: RoleService) {
        super();
        this.setHttp(http);
        this.setApi(environment.apiUrl + 'admin/proprietaire/');
    }

    public constrcutDto(): ProprietaireDto {
        return new ProprietaireDto();
    }

    public constrcutCriteria(): ProprietaireCriteria {
        return new ProprietaireCriteria();
    }
}

File: backend-ms1/src/main/java/ma/zs/easystock/service/facade/admin/gestionchequeffet/ProprietaireAdminService.java
package ma.zs.easystock.service.facade.admin.gestionchequeffet;

import java.util.List;
import ma.zs.easystock.bean.core.gestionchequeffet.Proprietaire;
import ma.zs.easystock.dao.criteria.core.gestionchequeffet.ProprietaireCriteria;
import ma.zs.easystock.zynerator.service.IService;



public interface ProprietaireAdminService extends  IService<Proprietaire,ProprietaireCriteria>  {
    Proprietaire findByUsername(String username);
    boolean changePassword(String username, String newPassword);




}
